src/case.f90

```diff
@@ -233,58 +233,73 @@ end subroutine preprocessing
   subroutine postprocessing(rho1, ux1, uy1, uz1, pp3, phi1, ep1)
 
     use decomp_2d, only : mytype, xsize, ph1
-    use visu, only  : write_snapshot, end_snapshot
-    use stats, only : overall_statistic
-
     use var, only : nzmsize
-    use var, only : itime
     use var, only : numscalar, nrhotime, npress
 
-    use turbine, only : turbine_output
-    use probes, only : write_probes
-
     real(mytype),dimension(xsize(1),xsize(2),xsize(3)), intent(in) :: ux1, uy1, uz1
     real(mytype),dimension(xsize(1),xsize(2),xsize(3),numscalar), intent(in) :: phi1
     real(mytype),dimension(xsize(1),xsize(2),xsize(3),nrhotime), intent(in) :: rho1
     real(mytype),dimension(xsize(1),xsize(2),xsize(3)), intent(in) :: ep1
     real(mytype),dimension(ph1%zst(1):ph1%zen(1), ph1%zst(2):ph1%zen(2), nzmsize, npress), intent(in) :: pp3
 
     integer :: j
-    integer :: num
     real(mytype),dimension(xsize(1),xsize(2),xsize(3),numscalar) :: T ! FIXME This can be huge
 
-    T = zero
-
     ! Recover temperature when decomposed (pressure to be recovered externally)
     if (itype.eq.itype_abl.and.ibuoyancy.eq.1) then
       do j=1,xsize(2) 
         T(:,j,:,1) = phi1(:,j,:,1) + Tstat(j,1)
       enddo
+      call real_postprocessing(rho1, ux1, uy1, uz1, pp3, T, ep1)
     else
-      T = phi1
+      call real_postprocessing(rho1, ux1, uy1, uz1, pp3, phi1, ep1)
     endif
 
+  end subroutine postprocessing
+  !##################################################################
+  !##################################################################
+  subroutine real_postprocessing(rho1, ux1, uy1, uz1, pp3, phi1, ep1)
```

Comment:
I'd suggest "run" postprocessing (or something like that), "real" could be misinterpreted as being related to the data type.

However, and I know I tend to argue against them, but would a pointer be the easier/neater solution here?

```
if (itype.eq.itype_able.and.ibuoyancy.eq.1) then
  do ... end do
  output_scalar => T
else
  output_scalar => phi1
end if

call write_snapshot(rho1, ux1, uz1, pp3, output_scalar, ep1, itime, num)
...
```